indexing-service/src/main/java/org/apache/druid/indexing/input/DruidInputSource.java

```diff
@@ -484,6 +501,54 @@ private static SortedMap<WindowedSegmentId, Long> createWindowedSegmentIdFromTim
     return segmentSizeMap;
   }
 
+
+  public List<TimelineObjectHolder<String, DataSegment>> getLockedTimelineForInterval(
+      RetryPolicyFactory retryPolicyFactory,
+      String dataSource,
+      Interval interval
+  )
+  {
+    Preconditions.checkNotNull(interval);
+
+    // This call used to use the TaskActionClient, so for compatibility we use the same retry configuration
+    // as TaskActionClient.
+    final RetryPolicy retryPolicy = retryPolicyFactory.makeRetryPolicy();
+    Collection<DataSegment> usedSegments;
+    while (true) {
+      try {
+        usedSegments = toolbox.getTaskActionClient()
```

Comment:
If you need to use only the `TaskActionClient`, you could just do a `@JacksonInject` of `TaskActionClientFactory` or `TaskMaster` in this class instead of setting `TaskToolbox` everywhere.

Reply:
Going by the comment above, this code might have used `TaskActionClient` earlier, you could also check how it was originally being done.